frontend/src/auth/TokenValidator.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
```

Comment:
We're wrapping some app-wide logic into a separate component, but the component itself doesn't render anything. It feels like this logic should be tied to the `App` component. I this what people usually do? I'm happy to accept it if so, it just looks a bit smelly to me.

Reply:
> Is this what people usually do?

I don't really know what people usually do for something like this.  I agree that it does seem a bit smelly but I feel that it's pretty harmless and does well to encapsulate things.